APCAppCore/APCAppCore/DataMonitor/APCDataMonitor+Bridge.m

```diff
@@ -39,32 +39,47 @@
 
 @implementation APCDataMonitor (Bridge)
 
-- (void) refreshFromBridgeOnCompletion: (void (^)(NSError * error)) completionBlock
+- (void) refreshFromBridgeOnCompletion: (APCDataMonitorResponseHandler) completionBlock
 {
-    if (self.dataSubstrate.currentUser.isConsented) {
-        [APCSchedule updateSchedulesOnCompletion:^(NSError *error) {
-            if (!error) {
-                [self.scheduler updateScheduledTasksIfNotUpdatingWithRange:kAPCSchedulerDateRangeToday];
-                [self.scheduler updateScheduledTasksIfNotUpdatingWithRange:kAPCSchedulerDateRangeTomorrow];
-                [APCTask refreshSurveysOnCompletion:^(NSError *error) {
-                    dispatch_async(dispatch_get_main_queue(), ^{
-                        [[NSNotificationCenter defaultCenter] postNotificationName:APCUpdateActivityNotification object:self userInfo:NULL];
-                    });
-                    if (completionBlock) {
-                        completionBlock(error);
-                    }
-                }];
+    if (self.dataSubstrate.currentUser.isConsented)
+    {
+        [[APCScheduler defaultScheduler] fetchTasksAndSchedulesFromServerAndThenUseThisQueue: [NSOperationQueue mainQueue]
+                                                                            toDoThisWhenDone: ^(NSError *errorFromServerFetch)
+         {
+             if (errorFromServerFetch)
+             {
+                 [self finishRefreshCommandPassingError: errorFromServerFetch
+                                      toCompletionBlock: completionBlock];
+             }
```

Comment:
Declare a weakSelf outside this block and use it here to avoid reference loop. 
`__weak typeof(self) weakSelf = self;`